src/drivers/dshot/DShot.cpp

```diff
@@ -532,12 +539,90 @@ void DShot::Run()
 		}
 	}
 
+	handle_vehicle_commands();
+
 	// check at end of cycle (updateSubscriptions() can potentially change to a different WorkQueue thread)
 	_mixing_output.updateSubscriptions(true);
 
 	perf_end(_cycle_perf);
 }
 
+void DShot::handle_vehicle_commands()
+{
+	vehicle_command_s vehicle_command;
+
+	while (!_current_command.valid() && _vehicle_command_sub.update(&vehicle_command)) {
+
+		if (vehicle_command.command == vehicle_command_s::VEHICLE_CMD_CONFIGURE_ACTUATOR) {
+			int function = (int)(vehicle_command.param5 + 0.5);
+
+			if (function < 1000) {
+				const int first_motor_function = 1; // from MAVLink ACTUATOR_OUTPUT_FUNCTION
+				const int first_servo_function = 33;
+
+				if (function >= first_motor_function && function < first_motor_function + actuator_test_s::MAX_NUM_MOTORS) {
+					function = function - first_motor_function + actuator_test_s::FUNCTION_MOTOR1;
+
+				} else if (function >= first_servo_function && function < first_servo_function + actuator_test_s::MAX_NUM_SERVOS) {
+					function = function - first_servo_function + actuator_test_s::FUNCTION_SERVO1;
+
+				} else {
+					function = INT32_MAX;
+				}
+
+			} else {
+				function -= 1000;
+			}
+
+			int type = (int)(vehicle_command.param1 + 0.5f);
```

Comment:
I guess this is to handle ACTUATOR_CONFIGURATION enum being translated into floating point can have value changes (e.g. 1 -> 0.999f), and to prevent that from being interpreted wrongly.

However, shouldn't we have some other generic function to handle 'pick closest integer' or so? Which can be just what's written above, but would be more verbose?